linodecli/configuration/__init__.py

```diff
@@ -286,31 +203,65 @@ def plugin_get_value(self, key):
 
         return self.config.get(username, full_key)
 
-    def write_config(self, silent=False):
+    # TODO: this is more of an argparsing function than it is a config function
+    # might be better to move this to argparsing during refactor and just have
+    # configuration return defaults or keys or something
+    def update(self, namespace, allowed_defaults):
+        """
+        This updates a Namespace (as returned by ArgumentParser) with config values
+        if they aren't present in the Namespace already.
+        """
+        if self.used_env_token and self.config is None:
+            return None
+        username = self.username or self.default_username()
+        if (not self.config.has_option(username, "token")
+            and not os.environ.get(ENV_TOKEN_NAME, None)):
+            print(f"User {username} is not configured.")
+            sys.exit(1)
+        if (not self.config.has_section(username)
+            or allowed_defaults is None):
+            return namespace
+
+        warn_dict = {}
+        ns_dict = vars(namespace)
+        for key in allowed_defaults:
+            if key not in ns_dict:
+                continue
+            if ns_dict[key] is not None:
+                continue
```

Comment:
Just double check, is this intent to continue when `key` is in `ns_dict` and the `value` is `None`?

Reply:
yes the ns_dict is the command arguments passed with the command, if it exists in ns_dict then we don't want to overwrite the custom setting with the default.